common/ferrostar/src/navigation_controller/mod.rs

```diff
@@ -100,6 +105,12 @@ impl NavigationController {
                         step: current_step,
                         linestring,
                     } => {
+                        let current_geometry_index = index_of_closest_origin_point(
+                            *snapped_user_location,
+                            &self.route.get_linestring(),
```

Comment:
I think this code is wrong? Why do we look at the entire route linestring here? This should always be over the current step IIUC.

Reply:
(To clarify why I think it's wrong, we usually look only at the current step both for efficiency and to limit weird step jumps like you or someone else mentioned is possible in Mapbox code. Other code calling the `index_of_closest_origin_point` function also limits to the current step.)